### ISSUE TITLE
ConvolverNode

### ISSUE THREAD

**Issue by bthj** [ircam-ismm/node-web-audio-api#13]
Hi 

`ConvolverNode` doesn't seem to be mapped.

I tried simply adding it to the [supportedNodes](https://github.com/ircam-ismm/node-web-audio-api/blob/main/generator/index.mjs#L10) array, but building after generating results in the error: 

```
error[E0596]: cannot borrow `*node` as mutable, as it is behind a `&` reference
   --> src/convolver_node.rs:331:5
    |
328 |     let node = napi_node.unwrap();
    |         ---- consider changing this binding's type to be: `&mut web_audio_api::node::ConvolverNode`
...
331 |     node.set_normalize(value);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^ `node` is a `&` reference, so the data it refers to cannot be borrowed as mutable
```

Can this generated code somehow be controlled to fix this error?

**Comment by orottier** [comment]
This is an upstream issue, thanks for flagging.
The `set_normalize` method should not require a mutable reference to self. We will have to fix this in a next version

**Comment by orottier** [comment]
This is fixed in web-audio-api-rs v0.26.0

**Comment by bthj** [comment]
Now when adding ConvolverNode with web-audio-api-rs v0.26.0, like so:

https://github.com/ircam-ismm/node-web-audio-api/compare/main...bthj:node-web-audio-api:feature/convolver-node

building results in the error:
```
error[E0308]: mismatched types
   --> src/audio_context.rs:108:22
    |
108 |             sink_id: None,
    |                      ^^^^ expected struct `std::string::String`, found enum `Option`
    |
    = note: expected struct `std::string::String`
                 found enum `Option<_>`

```

**Comment by orottier** [comment]
v0.26 has breaking changes, one of them is that the sink_id needs a string value.
You can use `String::new()` for the default audio output device to make it work again, but the best option would be to build full support for the sink_id parameter in the lib!

**Comment by bthj** [comment]
Thanks @orottier, it builds with `String::new()` and now I also tried to add bindings for the sink_id parameter:

https://github.com/ircam-ismm/node-web-audio-api/compare/main...bthj:node-web-audio-api:feature/convolver-node

- just not sure if this approach makes sense, @b-ma ?

**Comment by orottier** [comment]
I realize now that sink_id feature will only make sense when you bind [enumerate_devices](https://docs.rs/web-audio-api/latest/web_audio_api/fn.enumerate_devices.html) as well. That is maybe done best in a separate issue

**Comment by bthj** [comment]
In any case, I opened PR https://github.com/ircam-ismm/node-web-audio-api/pull/14

**Comment by b-ma** [comment]
Hey, yup that' looks good to me, thanks!

I will try to make a release with your fix soon, for the `enumerateDevices`, maybe it will have to wait a bit because I would prefer to implement it as a kind of reduced version of the `MediaDevices` API (cf. https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices), considering access to the microphone too (see #15)

**Comment by b-ma** [comment]
This should be released in 0.6.0, let's continue on #15 for the `enumerateDevice`question
